packages/regenesis-surgery/scripts/surgery.ts

```diff
@@ -0,0 +1,402 @@
+import { ethers } from 'ethers'
+import { KECCAK256_RLP_S, KECCAK256_NULL_S } from 'ethereumjs-util'
+
+interface Account {
+  address: string
+  nonce: number
+  balance: string
+  codeHash: string
+  root: string
+  code?: string
+  storage?: {
+    [key: string]: string
+  }
+}
+
+type StateDump = Account[]
+
+enum AccountType {
+  EOA,
+  PRECOMPILE,
+  PREDEPLOY_DEAD,
+  PREDEPLOY_WIPE,
+  PREDEPLOY_NO_WIPE,
+  PREDEPLOY_ETH,
+  PREDEPLOY_WETH,
+  DEAD,
+  UNISWAP_V3_FACTORY,
+  UNISWAP_V3_NFPM,
+  UNISWAP_V3_POOL,
+  UNISWAP_V3_LIB,
+  UNISWAP_V3_OTHER,
+  UNVERIFIED,
+  VERIFIED,
+}
+
+interface UniswapPoolData {
+  oldAddress: string
+  newAddress: string
+  token0: string
+  token1: string
+  fee: ethers.BigNumber
+}
+
+interface SurgeryDataSources {
```

Comment:
This also needs the etherscan contracts dump

Reply:
https://github.com/ethereum-optimism/optimism/pull/1488/files#diff-d50b4c54e0bcc06b809e8c85ac35f94e5ab7b8ac6c61695a84415b55a27d2d15R43